sd/etcd/registrar.go

```diff
@@ -19,9 +25,27 @@ type Registrar struct {
 type Service struct {
 	Key           string // unique key, e.g. "/service/foobar/1.2.3.4:8080"
 	Value         string // returned to subscribers, e.g. "http://1.2.3.4:8080"
+	TTL           *TTLOption
 	DeleteOptions *etcd.DeleteOptions
 }
 
+// TTLOption allow setting a key with a TTL, and regularly refreshes the lease with a goroutine
+type TTLOption struct {
+	Heartbeat time.Duration
+	TTL       time.Duration
+}
+
+// NewTTLOption returns a TTLOption
+func NewTTLOption(heartbeat, ttl time.Duration) *TTLOption {
```

Comment:
The doc comment here is a little sparse. Could it please describe what each of the parameters are, and some good default values? It would also be good to describe the bounds-checking logic implemented in the function body.